### DISCUSSION TITLE
2.4" I2C Display

### DISCUSSION THREAD

**Discussion by Jens869** [tbnobody/OpenDTU#1116] [General]
es werden ja SSD1306 und SH1106 Displays mit einer Auflösung von 128*64 unterstützt.
Welchen Displaytyp haben SSD1309 Displays mit 2,42" mit 128*64 Auflösung ?

durch umlöten von zwei Widerständen verstehen die auch I2C und haben das selbe Pinout wie 1,3" Displays

**Comment by BastyESP** [comment]
Auch wenn ich eher aus dem AhoyDTU Lager komme (hab aber auch nebenher eine OpenDTU am laufen nur da nicht das 2.42" angesteckt) und dort ein 2.42" am Laufen habe, kann ich dir nur das "Schwarze" 2.42" als 4 Pin ---> I²C empfehlen, dort D2 kurzschließen und es Läuft, sicherlich auch mit der OpenDTU. 
Ob es mit dem Blauen auch funktioniert werden dir andere sagen können, meine Erfahrungen mit dem Schwarzen laufen nun schon gute 4 wochen, Problemlos, musst es als 0.96" ansprechen, dann Funktioniert es.

Laut einem User geht auch noch das 1.54" OLED
https://github.com/lumapu/ahoy/issues/1012 les dir das am besten mal durch...ist zwar Ahoy aber die sind sich ja nicht Total unterschiedlich.

**Reply by smdfritz** [reply]
Hallo

Ich habe ebenfall ein 2.42" Display hier vor mir, aber mit 7 pins. Wie bekomme ich es zum Laufen? Auf der Rückseite steht was von I2C mit R3 und R5. Aber falls das geht, welche Widerstände kommen dort rein? Oder geht es auch im SPI Modus?
![IMG_20250220_191416](https://github.com/user-attachments/assets/359cf9df-260c-4e20-8adf-0cf55cfca0f4). Mit Type "4" und CS/RESET Zuweisung der Pins habe ich es schon probiert, läuft aber nicht. Was mache ich falsch?

        },
        "display": {
            "type": 4,
            "data": 21,
            "clk": 22,
            "cs" : 12,
            "reset" : 13
        }

**Comment by Jens869** [comment]
Wenn R3, R5 bestückt sind ist es I2C. Ist nur R4 drauf dann SPI. SCL ist Clock und SDA ist Data in. Wenn CS Low ist, dann regiert es auf Daten.

**Reply by smdfritz** [reply]
Also auf R3 und R5 auch einen 4,7k und dann sollte es laufen? Muss ich über den ESP32 die Pins CS und RESET überhaupt verbinden oder genügen dann die vier Standard VCC,GND,SCL und SDA?

**Comment by smdfritz** [comment]
Habe es zum Laufen bekommen. R4 auf R3 umsetzen und R5 mit 0 Ohm überbrücken. CS und RESET müssen im Pinmapping vergeben werden. Typ 2 und 4 laufen wahlweise.
